lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -43,7 +44,47 @@ function runA11yChecks() {
       'blink': {enabled: false},
       'server-side-image-map': {enabled: false}
     }
+  }).then(axeResult => {
+    // Augment the node objects with outerHTML snippet & custom path string
+    axeResult.violations.forEach(v => v.nodes.forEach(node => {
+      node.path = getNodePath(node.element);
+      node.snippet = getOuterHTMLSnippet(node.element);
+    }));
+
+    return axeResult;
   });
+
+  // Adapated from DevTools' SDK.DOMNode.prototype.path
+  //   https://github.com/ChromeDevTools/devtools-frontend/blob/7a2e162ddefd/front_end/sdk/DOMModel.js#L530-L552
+  // TODO: Doesn't handle frames or shadow roots...
+  function getNodePath(node) {
+    /* global Node */
+    function getNodeIndex(node) {
+      let index = 0;
+      while (node = node.previousSibling) {
+        // skip empty text nodes
+        if (node.nodeType === Node.TEXT_NODE &&
+            node.textContent.trim().length === 0) continue;
+        index++;
+      }
+      return index;
+    }
+
+    const path = [];
+    while (node && node.parentNode) {
+      const index = getNodeIndex(node);
+      path.push([index, node.nodeName]);
```

Comment:
yeah [in the protocol docs](https://chromedevtools.github.io/devtools-protocol/tot/DOM/#method-pushNodeByPathToFrontend) it says "proprietary format". :)

impl's: [backend side](https://cs.chromium.org/chromium/src/third_party/WebKit/Source/core/inspector/InspectorDOMAgent.cpp?type=cs&q=pushNodeByPathToFrontend&sq=package:chromium&l=2063),[frontend side](https://github.com/ChromeDevTools/devtools-frontend/blob/7a2e162ddefd/front_end/sdk/DOMModel.js#L530-L552)

downsides of using selectors:

1. can't reference text nodes, document fragments, or the doctype node
1. can't pierce shadow dom